labs/bdd-proxy-development/lab.md

```diff
@@ -0,0 +1,210 @@
+
+### Environment Variables
+
+Commands used during the lab will require the following environment variables
+to be configured.
+
+``` bash
+export TOKEN=$(gcloud auth application-default print-access-token)
+export APIGEE_ENV=<apigee-environment-name>
+export APIGEE_ORG=<gcp-project-name>
+```
+
+### oas-apigee-mock
+
+Clone the Apigee DevRel repo and install the dependencies for the
+oas-apigee-mock tool.
+
+``` bash
+git clone https://github.com/apigee/devrel.git
+cd devrel/tools/oas-apigee-mock/
+export LAB_HOME=$PWD
+npm install
+```
+
+## Generate the proxy bundle and prepare tests
+
+Duration: 0:10:00
+
+### Generate an Apigee proxy from an Open API Specification
+
+Use the `orders-apikey-header.yaml` Open API specification included
+in the `$LAB_HOME/test/oas` folder to generate a proxy bundle.
+
+``` bash
+node $LAB_HOME/bin/oas-apigee-mock generateApi web-orders-proxy-v1 \
+  -s $LAB_HOME/test/oas/orders-apikey-header.yaml \
+  -o
+```
+
+### Update the apickli configuration to use your Environment Group's Hostname
+
+Update the `before(function ()` block in
+`$LAB_HOME/test/features/step_definitions/init.js` with a hostname from the
+environment group you will deploying to. For example:
+
+``` javascript
+before(function () {
+  this.apickli = new apickli.Apickli(
+    "https",process.env.RUNTIME_HOST_ALIAS
+  );
+```
+
+If you followed the
+[Apigee X Trial Provisioning](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+script, a hostname will have been stored in the `RUNTIME_HOST_ALIAS`
+enviornment variable associated with the `eval-group` environment group.
+
+You can also retrive environment group information from you Apigee organisation
+using `apigeecli`.
+
+``` bash
+apigeecli envgroups list -t "$TOKEN" -o "$APIGEE_ORG"
+```
+
+Using `jq` we can retrieve a hostname of an environment group and store it in the
+`RUNTIME_HOST_ALIAS` variable.
+
+``` bash
+apigeecli envgroups list -t "$TOKEN" -o "$APIGEE_ORG" > environment-groups.json
+RUNTIME_HOST_ALIAS=$(jq '.environmentGroups[0].hostnames[0]' -r < environment-groups.json )
+export RUNTIME_HOST_ALIAS
+echo "RUNTIME_HOST_ALIAS is $RUNTIME_HOST_ALIAS"
+```
+
+### Update the test suite proxy reference
+
+Update the test suite `$LAB_HOME/test/features/orders-apikey-header.feature`
+with the basepath for your proxy.`web-orders-proxy-v1` was used in the generate
+proxy step earlier in this lab.
+
+## Deploy and Test the Proxy
+
+Duration: 0:10:00
+
+### Test the Proxy
+
+Lets start by running the test suite.
+
+``` bash
+cd $LAB_HOME
+./node_modules/.bin/cucumber-js test/features/orders-apikey-header.feature --format json:test/test_report.json --publish-quiet
+```
+
+The test should fail as we have yet to deploy the proxy we generated
+previously.
+
+### Deploy the generated proxy to your Apigee Organisation
+
+``` bash
+sackmesser deploy -d "$LAB_HOME/api_bundles/web-orders-proxy-v1" --googleapi -t "$TOKEN" -o "$APIGEE_ORG" -e "$APIGEE_ENV"
```

Comment:
@markjkelly to reduce the number of different tools you might want to consider doing this via the apigeecli tool as well. I'll leave it up to you.

Reply:
+1 but not a blocker